src/test/regress/sql/generated_identity.sql

```diff
@@ -0,0 +1,266 @@
+
+SELECT alter_distributed_table('generated_identities_test', 'c');
+
+SELECT undistribute_table('generated_identities_test');
+
+SELECT * FROM generated_identities_test ORDER BY g;
+
+-- reference table
+
+DROP TABLE generated_identities_test;
+
+CREATE TABLE generated_identities_test (
+    a int GENERATED BY DEFAULT AS IDENTITY,
+    b bigint GENERATED ALWAYS AS IDENTITY (START WITH 10 INCREMENT BY 10),
+    c smallint GENERATED BY DEFAULT AS IDENTITY,
+    d serial,
+    e bigserial,
+    f smallserial,
+    g int
+);
+
+SELECT create_reference_table('generated_identities_test');
+
+\d generated_identities_test
+
+\c - - - :worker_1_port
+
+\d generated_identities.generated_identities_test
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO generated_identities_test (g)
+SELECT s FROM generate_series(11,20) s;
+
+SELECT * FROM generated_identities_test ORDER BY g;
+
+SELECT undistribute_table('generated_identities_test');
+
+\d generated_identities_test
+
+\c - - - :worker_1_port
+
+\d generated_identities.generated_identities_test
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+-- alter table .. add column .. GENERATED .. AS IDENTITY
+DROP TABLE IF EXISTS color;
+CREATE TABLE color (
+    color_name VARCHAR NOT NULL
+);
+SELECT create_distributed_table('color', 'color_name');
+ALTER TABLE color ADD COLUMN color_id BIGINT GENERATED ALWAYS AS IDENTITY;
+INSERT INTO color(color_name) VALUES ('Red');
+ALTER TABLE color ADD COLUMN color_id_1 BIGINT GENERATED ALWAYS AS IDENTITY;
+DROP TABLE color;
+
+-- insert data from workers
+CREATE TABLE color (
+    color_id BIGINT GENERATED ALWAYS AS IDENTITY UNIQUE,
+    color_name VARCHAR NOT NULL
+);
+SELECT create_distributed_table('color', 'color_id');
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+SELECT undistribute_table('color');
+SELECT create_distributed_table('color', 'color_id');
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+SELECT count(*) from color;
+
+-- modify sequence & alter table
+DROP TABLE color;
+
+CREATE TABLE color (
+    color_id BIGINT GENERATED ALWAYS AS IDENTITY UNIQUE,
+    color_name VARCHAR NOT NULL
+);
+SELECT create_distributed_table('color', 'color_id');
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+\c - - - :master_port
```

Comment:
for future reference, our test files nowadays often avoid doing a lot of back-and-forth connecting (e.g. by using run_command_on_workers), because that enables the test file against a local Citus cluster (in case of issues) or run via arbitrary configs test. It also makes the test somewhat faster, since every \c starts a new session. There are a lot of test files that ignore this because we don't actively clean up our tests.